pulsar-io/jdbc/clickhouse/pom.xml

```diff
@@ -38,7 +38,7 @@
       <version>${project.version}</version>
     </dependency>
     <dependency>
-      <groupId>ru.yandex.clickhouse</groupId>
+      <groupId>com.clickhouse</groupId>
```

Comment:
Perhaps it's better to use shaded jar?

<dependency>
    <groupId>com.clickhouse</groupId>
    <artifactId>clickhouse-jdbc</artifactId>
    <version>0.3.2-patch11</version>
    <classifier>all</classifier>
    <exclusions>
        <exclusion>
            <groupId>*</groupId>
            <artifactId>*</artifactId>
        </exclusion>
    </exclusions>
</dependency>

Reply:
Thanks for your review @zhicwu!

What's the most common dependency conflict that can be avoided by using the shaded jar?

I can see using the basic JAR, the final connector NAR is in size 9.7M, and after switching to the shaded jar, it grows to 46M.

If no user reports the current solution's issue, I tend to keep it as is.

Reply:
> What's the most common dependency conflict that can be avoided by using the shaded jar?

LZ4 and Gson. Put native library aside, if there's custom classloader for isolation, it should not be a problem.

> I can see using the basic JAR, the final connector NAR is in size 9.7M, and after switching to the shaded jar, it grows to 46M.

Did you exclude all dependencies when specifying classifier? I'm not familiar with the Nifi maven plugin, but the JDBC driver with only http implementation is ~1MB. `-all` contains more like gRPC etc., so it's 20+ MB.

Reply:
> Did you exclude all dependencies when specifying classifier?

No. I'm unsure what dependencies should be excluded XD.

Reply:
All of them because you only need the shaded jar :)

```xml
<dependency>
    <groupId>com.clickhouse</groupId>
    <artifactId>clickhouse-jdbc</artifactId>
    <version>0.3.2-patch11</version>
    <classifier>all</classifier>
    <exclusions>
        <exclusion>
            <groupId>*</groupId>
            <artifactId>*</artifactId>
        </exclusion>
    </exclusions>
</dependency>
```

Reply:
Now it reduces to 25M